R/get_connection.R

```diff
@@ -199,3 +125,77 @@ credd_from_local <- function(){
   })
   return(c(username, password))
 }
+
+#' check if the environment is local or in nuvolos
+is_local <- function(){
+  path_filename <- Sys.getenv("ACLIB_DBPATH_FILE", "/lifecycle/.dbpath")
+  local = FALSE
+  if (!file.exists(path_filename)){
+    local = TRUE
+  } 
+  return(local)
+}
+
+get_local_info <- function(username = NULL, password = NULL, dbname = NULL, schemaname = NULL){
+  # Find credentials from local, if not, create one by keyring
+  if (is.null(username) && is.null(password)) {
+    tryCatch({
+      cred = credd_from_local()
+    }, error = function(e) {
+      input_nuvolos_credential()
+      cred = credd_from_local()
+    })
+    username = cred[1]
+    password = cred[2]
+  } else if(is.null(username) && !is.null(password) ) {
+    stop("Inconsistent input: username is missing, but password was provided. Please specify a username or leave both arguments as NULL.")
+  } else if(!is.null(username) && is.null(password) ) {
+    stop("Inconsistent input: password is missing, but username was provided. Please specify a password or leave both arguments as NULL.")
+  }
+  
+  ## DB & SCHEMA ##
+  # Only accept dbname and schemaname together. If both are missing resort to relying on get_nuvolos_db_path() to figure it out. 
+  if (is.null(dbname) | is.null(schemaname)) {
+    stop("Inconsistent input: please provide both dbname and schemaname.")
+  }
+  
+  return(list(username, password, dbname, schemaname))
+}
+
+get_nuvolos_info <- function(username = NULL, password = NULL, dbname = NULL, schemaname = NULL){
+  # Relying on get_nuvolos_db_path() to find db and schema. 
+  if (is.null(dbname) && is.null(schemaname)) {
+    conn_info <- get_nuvolos_db_path()
+    dbname <- conn_info$dbname
+    schemaname <- conn_info$schemaname
+  } 
```

Comment:
we should have an else if (is.null(dbname) ||  is.null(schemaname)) to make sure that if at least one is specified, then both are (similarly like for local connector)